### ISSUE TITLE
Portal default timezone settings should be taken into account for effective/expiration dates

### ISSUE THREAD

**Issue by agitator** [plone/Products.CMFPlone#1298]

**Comment by rileydog** [comment]
Please see related issue #1290 .  changes to time zone in site config don't stick.

**Comment by agitator** [comment]
any pointers about the reason or update on this?

**Comment by tkimnguyen** [comment]
I've had two people report this problem this week... should either show the dates and times using the browser's timezone or should display the timezone of the server and site. Use something like this? 

```
<span tal:content="python:view.toLocalizedTime(item['time'],long_format=True)" class="pat-moment" data-pat-moment="format:relative;"/>
```

**Comment by thet** [comment]
I would align the effective/expiration date to the behavior of plone.app.event, where much timezone mangling was done.
plone.app.event stores it's start/end date as local time with timezone information. their indexing values are converted to UTC first, though.
the same should be done with effective/expiration dates.

historical note: before plone.app.event 2.0 we stored start/end times as UTC values. but that's less flexible (e.g. start and end had to be in the same timezone) and not needed. date/time calculations with different timezones are done correctly by python itself.

**Comment by thet** [comment]
Also got this problem reported by a client.
This needs a fix - effective/expires should be set in the portal's timezone, as Python datetime objects.

**Comment by thet** [comment]
As I stumbled over that problem again I took a look about the possible complexity of this task.
Could actually be easy. Not for Plone 5.1 as a RC1 is going to be released soon. Better Plone 5.2+.
So:

This is how a timezone aware datetime schema field is defined - https://github.com/plone/plone.app.event/blob/master/plone/app/event/dx/behaviors.py#L82
```
class IEventBasic(model.Schema, IDXEvent):
    start = schema.Datetime()
    directives.widget(
        'start',
        DatetimeFieldWidget,
        default_timezone=default_timezone
)
```

In plone.app.z3cform.widget the DatetimeWidget supports timezones, just by allowing the timezone attribute to be set: https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/widget.py#L190

Same package, converters - The DatetimeWidgetConverter toFieldValue method does the actual timezone handling - it converts the naive value from the widget into a timezone'd Python datetime value using the default_timezone: https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/converters.py#L107

So - in plone.app.dexterity.behaviors.metadata, the IPublication behavior's effective and expires attributes must be configured to support a default_timezone, like it's done in plone.app.event.
https://github.com/plone/plone.app.dexterity/blob/master/plone/app/dexterity/behaviors/metadata.py#L146

And - quite interesting and the key - the DCFieldProperty class' __get__ (https://github.com/plone/plone.app.dexterity/blob/master/plone/app/dexterity/behaviors/metadata.py#L300) and __set_ (https://github.com/plone/plone.app.dexterity/blob/master/plone/app/dexterity/behaviors/metadata.py#L324) methods must be changed so that there is no converting into a Zope DateTime any more.

**Comment by vangheem** [comment]
Yup, dexterity behavior strips tz info out of object: https://github.com/plone/plone.app.dexterity/blob/master/plone/app/dexterity/behaviors/metadata.py#L332

Then, when value is returned again, https://github.com/plone/plone.app.dexterity/blob/master/plone/app/dexterity/behaviors/metadata.py#L306

ouch...
